src/main/java/games/strategy/util/Util.java

```diff
@@ -122,4 +130,19 @@ public static String getStringFromInputStream(InputStream in) {
     }
     return builder.toString();
   }
+
+  /**
+   * Creates a hash of the given String based on the SHA-512 algorithm.
+   * 
+   * @param input The input String to hash.
+   * @return A hashed hexadecimal String of the input.
+   */
+  public static String sha512(String input) {
```

Comment:
There are several other occurences (in the future) where this method will be needed (for example for hashed macs, which are currently hashed with the exact same salt `MH` everywhere).
Adding unit tests for this method is going to be rather simple...